### PR TITLE
nv4x: boot option personalizations

### PR DESCRIPTION
**Pull request by mkopec**
draft until https://github.com/Dasharo/edk2/pull/20 lands

- add option to disable secureboot by default
- reduce boot timeout to 1 second

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

### CODE REVIEW

**Comment by mkopec** [comment]
@miczyg1 Do we just update the tianocore revision here? Doing so will also enable the Dasharo system options menu which won't work on laptops since the implementation on coreboot's side is missing. The menu should probably be hidden behind a config option

**Comment by miczyg1** [comment]
@mkopec yes, we should probably do so.

**Comment by mkopec** [comment]
Wired up the SecureBootDefaultEnable PCD to Kconfig.

**Comment by macpijan** [comment]
> draft until https://github.com/Dasharo/edk2/pull/20 lands

It landed, what's the nex step? Review from @miczyg1 ?

File: build.sh [hunk 1]
```diff
@@ -21,6 +21,7 @@ usage() {
   echo "   -l path_to_logo.bmp: Build with custom boot logo"
   echo "                    -s: Build with new vboot keys for testing vboot"
   echo "                    -f: Also build full flash image (requires me.bin and ifd.bin)"
+  echo "                    -u: Build with UEFI Secure Boot disabled"
```
Comment (miczyg1) [inline build.sh:24]:
```suggestion
  echo "                    -u: Build with UEFI Secure Boot default state to disabled"
```

File: build.sh [hunk 2]
```diff
@@ -88,6 +99,9 @@ build() {
   if [[ $REPLACE_KEYS = 1 ]]; then
     replace_keys
   fi
+  if [[ $NO_SECURE_BOOT = 1 ]]; then
```
Comment (miczyg1) [inline build.sh:102]: NO_SECURE_BOOT is confusing as it does not do what it means

File: payloads/external/tianocore/Kconfig [hunk 3]
```diff
@@ -200,85 +199,15 @@ config TIANOCORE_SECURE_BOOT
...
-
-
-config TIANOCORE_ENABLE_IPXE
-	bool "Include iPXE in Tianocore payload"
+config TIANOCORE_SECURE_BOOT_DEFAULT_ENABLE
+	bool "Enable Secure Boot by default"
 	default y
 	help
-	  Includes a prebuilt iPXE for EFI to the TianoCore payload.
+	  Enables UEFI Secure Boot by default.
```
Comment (miczyg1) [inline payloads/external/tianocore/Kconfig:208]:
```suggestion
	  Sets the UEFI Secure Boot state to enabled by default.
```

File: payloads/external/tianocore/Kconfig [hunk 4]
```diff
@@ -200,85 +199,15 @@ config TIANOCORE_SECURE_BOOT
...
-	  F10        0x0014
-	  F11        0x0015
-	  F12        0x0016
-	  ESC        0x0017
-
-
-config TIANOCORE_ENABLE_IPXE
-	bool "Include iPXE in Tianocore payload"
+config TIANOCORE_SECURE_BOOT_DEFAULT_ENABLE
+	bool "Enable Secure Boot by default"
```
Comment (miczyg1) [inline payloads/external/tianocore/Kconfig:205]:
```suggestion
	bool "Set Secure Boot state to enabled by default"
```

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: build.sh (hunk 2 above)
Comment (mkopec) [reply build.sh]: Changed to SECURE_BOOT_DEFAULT_DISABLED

**Comment by mkopec** [comment]
> Address review is probably not the best commit name

Commit message updated.

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**